File: app/src/main/java/com/group15/fitnesstracker/db/MicronutrientRecording.java
package com.group15.fitnesstracker.db;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;


@Entity(foreignKeys = @ForeignKey(entity = NutritionRecording.class,
                                  parentColumns = "recording_id",
                                  childColumns = " recording_id"))
public class MicronutrientRecording {
    @PrimaryKey
    public int recordingId;

    public String name;

    public double amount;
}

File: app/src/main/java/com/awave/wrkout/db/MicronutrientRecording.kt
package com.awave.wrkout.db

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey


@Entity(foreignKeys = [ForeignKey(entity = NutritionRecording::class, parentColumns = ["recordingId"], childColumns = ["recordingId"])])
data class MicronutrientRecording (
    @PrimaryKey(autoGenerate = true) var id: Int = 0,
    var recordingId: Int = 0,
    var name: String? = null,
    var amount: Double = 0.toDouble()
)
